sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -467,13 +468,13 @@ case class DataSourceStrategy(conf: SQLConf) extends Strategy with Logging with
   }
 }
 
-object DataSourceStrategy {
+object DataSourceStrategy extends PredicateHelper {
   /**
    * The attribute name may differ from the one in the schema if the query analyzer
    * is case insensitive. We should change attribute names to match the ones in the schema,
    * so we do not need to worry about case sensitivity anymore.
    */
-  protected[sql] def normalizeExprs(
```

Comment:
not sure if this is needed given `HiveStrategies.scala` is also within the `sql` scope?

Reply:
ah, this is to follow https://github.com/apache/spark/pull/29831#discussion_r499528767 as we are in execution package, we could eliminate these per-method package-level constraint?

Reply:
yeah im not sure about the comment - perhaps @HyukjinKwon meant to replace `private[sql]` with `private` (which won't work since this is used by Hive. Otherwise it increases the scope to public which I'm not sure is a good thing.

Reply:
Probably I wasn't clear. I meant to just don't use `private[sql]`. since `execution` is already a private package.

Reply:
This was fixed at SPARK-16964

Reply:
Ah thanks @HyukjinKwon !